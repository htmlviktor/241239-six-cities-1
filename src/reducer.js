import {loadOffers} from './actions';

const initialState = {
  currentCity: `Amsterdam`,
  listCities: new Set([...data.map((offer) => offer.city)]),
  offers: []
};

const Operation = {
  loadOffers: () => (dispatch, _getState, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        dispatch(loadOffers(response.data));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CITY_CHANGE`:
      return Object.assign({}, state, {
        currentCity: action.city
      });
    case `LOAD_OFFERS`:
      return Object.assign({}, state, {
        offers: action.offers
      });
    default: return state;
  }
};


export {
  reducer,
  Operation
};
