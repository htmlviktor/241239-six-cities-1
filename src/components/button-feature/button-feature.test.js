import React from 'react';
import renderer from 'react-test-renderer';
import ButtonFeature from './button-feature.jsx';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../reducer/index';

it(`Test ButtonFeature Component`, () => {
  const tree = renderer
    .create(
        <Provider store={createStore(reducer)}>
          <ButtonFeature />
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
