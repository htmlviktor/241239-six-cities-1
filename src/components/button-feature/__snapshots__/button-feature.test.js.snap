// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonFeature Component 1`] = `
<button
  className="undefined button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
